index.js

```diff
@@ -48,13 +49,73 @@ module.exports = {
           ), self.findOrCreateUser(spec));
           spec.name = dummy.name;
         }
+        spec.label = spec.label || spec.name;
         spec.options.callbackURL = self.getCallbackUrl(spec, true);
         self.strategies[spec.name] = new Strategy(spec.options, self.findOrCreateUser(spec));
         self.apos.login.passport.use(self.strategies[spec.name]);
         self.addLoginRoute(spec);
         self.addCallbackRoute(spec);
         self.addFailureRoute(spec);
       });
+
+      self.on('apostrophe-login:after', 'redirectToNewLocale', async req => {
+        const workflow = self.apos.modules['apostrophe-workflow'];
+        if (!workflow) {
+          return;
+        }
+        const cacheSet = util.promisify(workflow.crossDomainSessionCache.set);
+        const deserializeUser = util.promisify(self.apos.login.deserializeUser);
+        if (!req.session.passportWorkflow) {
+          return;
+        }
+        const {
+          oldLocale,
+          newLocale,
+          oldSlug
+        } = req.session.passportWorkflow;
+        delete req.session.passportWorkflow;
+        const crossDomainSessionToken = self.apos.utils.generateId();
```

Comment:
Unfortunately the workflow module is not perfectly factored for this, thus some duplication with it, but that module is very late stage and we're not going to change the behavior incompatibly there.